docs/DeveloperGuide.md

```diff
@@ -1,22 +1,264 @@
-# Developer Guide
+# JavaNinja Developer Guide
+Java Ninja is a command-line interface (CLI) learning tool focused on helping beginner programmers enhance their understanding of fundamental programming concepts.
+
+- [Design & implementation](#design-&-implementation)
+- [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+    - [Value proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-Functional-Requirements)
+- [Glossary](#Glossary)
+- [Instructions for manual testing](#Instructions-for-manual-testing)
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- [Apache Commons Lang Library](https://commons.apache.org/proper/commons-lang/) - Used for string utilities.
+- [JUnit 5](https://junit.org/junit5/) - Used for testing.
+- Code adapted from [AddressBook-Level3](https://github.com/se-edu/addressbook-level3).
+
+JavaNinja uses the following tools for development:
+- JUnit - Used for testing.
+- Gradle - Used for build automation.
 
 ## Design & implementation
+  
+### Design 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Architecture 
+
+Main components of the architecture:
+
+`Main` (consisting of class `JavaNinja`) is in charge of the app launch and shut down. 
+1. At app launch, it initialises the components in the correct order and connects them with each other
```

Comment:
Perhaps there could have been more illustration to show the correct order of initialisation.